test/SlowTests/RavenDB-14303.cs

```diff
@@ -31,40 +31,46 @@ public async Task IgnoreIdleIndexOnPromotable()
+                        using (var session = store.OpenAsyncSession())
+                        {
+                            await session.StoreAsync(new User { Name = "Toli", Count = i, Age = rnd.Next(1, 100) }, "users/" + i);
+                            await session.SaveChangesAsync();
+                        }
                     }
-                }
-
-                var count = new AutoIndexField
-                {
-                    Name = "Count",
-                };
 
-                var age = new AutoIndexField
-                {
-                    Name = "Age",
-                };
+                    var count = new AutoIndexField
+                    {
+                        Name = "Count",
+                    };
 
-                var sum = new AutoIndexField
-                {
-                    Name = "Sum",
-                };
-                var node = nodes.First(n => n.ServerStore.DatabasesLandlord.IsDatabaseLoaded(store.Database));
-                var db = await node.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(store.Database);
-                var index = await db.IndexStore.CreateIndex(new AutoMapReduceIndexDefinition("Users", new[] { count, sum }, new[] { age }), Guid.NewGuid().ToString());
+                    var age = new AutoIndexField
+                    {
+                        Name = "Age",
+                    };
 
-                await leader.ServerStore.Engine.PutAsync(new SetIndexStateCommand(index.Name, IndexState.Idle, db.Name, Guid.NewGuid().ToString()));
-                db.ServerStore.ForTestingPurposesOnly().StopIndex = true;
+                    var sum = new AutoIndexField
+                    {
+                        Name = "Sum",
+                    };
+                    var node = nodes.First(n => n.ServerStore.DatabasesLandlord.IsDatabaseLoaded(store.Database));
+                    var db = await node.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(store.Database);
+                    var index = await db.IndexStore.CreateIndex(new AutoMapReduceIndexDefinition("Users", new[] { count, sum }, new[] { age }), Guid.NewGuid().ToString());
 
-                var addRes = await store.Maintenance.Server.SendAsync(new AddDatabaseNodeOperation(store.Database));
-                await WaitForRaftIndexToBeAppliedInCluster(addRes.RaftCommandIndex, TimeSpan.FromSeconds(5));
+                    await leader.ServerStore.Engine.PutAsync(new SetIndexStateCommand(index.Name, IndexState.Idle, db.Name, Guid.NewGuid().ToString()));
+                    db.ServerStore.ForTestingPurposesOnly().StopIndex = true;
 
+                    var addRes = await store.Maintenance.Server.SendAsync(new AddDatabaseNodeOperation(store.Database));
+                    await WaitForRaftIndexToBeAppliedInCluster(addRes.RaftCommandIndex, TimeSpan.FromSeconds(5));
+                }
+                catch (Exception e)
+                {
```

Comment:
The suspicion here is that some error occurred during the test and another error happened during the disposal of the document store, which masked the original exception. (which might be related or not)
 
This will not help you. You will throw without recording it -> dispose the store -> mask the exception.
Let's have a larger scope for the try, up to the dispose it self and do maybe `Console.WriteLine`